### PR TITLE
ui: more theming utilities (for Files)

### PR DESCRIPTION
**Pull request by AlvaroBrey**
Files PR: https://github.com/nextcloud/android/pull/10717

- [x] deduplicate code

### CODE REVIEW

**Review by AndyScherzinger — APPROVED** [review]
LGTM 👍

File: ui/src/main/java/com/nextcloud/android/common/ui/theme/utils/MaterialViewThemeUtils.kt [hunk 1]
```diff
@@ -65,39 +68,54 @@ class MaterialViewThemeUtils @Inject constructor(schemes: MaterialSchemes, priva
...
+            fab.imageTintList = ColorStateList(
+                arrayOf(
+                    intArrayOf(android.R.attr.state_enabled),
+                    intArrayOf(-android.R.attr.state_enabled)
+                ),
+                intArrayOf(
+                    scheme.onPrimaryContainer,
+                    scheme.onSurfaceVariant
+                )
+            )
```
Comment (AlvaroBrey) [inline ui/src/main/java/com/nextcloud/android/common/ui/theme/utils/MaterialViewThemeUtils.kt:91]: @AndyScherzinger you can use `buildColorStateList` to simplify this, for reference:

```suggestion
            fab.backgroundTintList = buildColorStateList(
                android.R.attr.state_enabled to scheme.primaryContainer,
                -android.R.attr.state_enabled to scheme.surfaceVariant
            )

            fab.imageTintList = buildColorStateList(
                android.R.attr.state_enabled to scheme.onPrimaryContainer,
                -android.R.attr.state_enabled to scheme.onSurfaceVariant
            )
```

File: ui/src/main/java/com/nextcloud/android/common/ui/theme/utils/MaterialViewThemeUtils.kt [hunk 2]
```diff
@@ -65,39 +68,54 @@ class MaterialViewThemeUtils @Inject constructor(schemes: MaterialSchemes, priva
...
+            fab.imageTintList = ColorStateList(
+                arrayOf(
+                    intArrayOf(android.R.attr.state_enabled),
+                    intArrayOf(-android.R.attr.state_enabled)
+                ),
+                intArrayOf(
+                    scheme.onPrimaryContainer,
+                    Color.WHITE
+                )
+            )
```
Comment (AlvaroBrey) [inline ui/src/main/java/com/nextcloud/android/common/ui/theme/utils/MaterialViewThemeUtils.kt:91]: @AndyScherzinger you can use `buildColorStateList` to simplify this, for reference:

```suggestion
            fab.backgroundTintList = buildColorStateList(
                android.R.attr.state_enabled to scheme.primaryContainer,
                -android.R.attr.state_enabled to Color.GRAY
            )

            fab.imageTintList = buildColorStateList(
                android.R.attr.state_enabled to scheme.onPrimaryContainer,
                -android.R.attr.state_enabled to Color.WHITE
            )
```
@AndyScherzinger  you can use `buildColorStateList` to simplify

**Comment by AndyScherzinger** [comment]
Woohoo! 💙

**Comment by AlvaroBrey** [comment]
> Woohoo! blue_heart

Thanks for all your contributions!

**Pull request merged**